docs/streaming_endpoints.md

```diff
@@ -0,0 +1,50 @@
+
+```
+service GRPCInferenceService
+{
+  ...
+  // Standard
+  rpc ModelInfer(ModelInferRequest) returns (ModelInferResponse) {}
+  // Extension
+  rpc ModelStreamInfer(stream ModelInferRequest) returns (stream ModelStreamInferResponse) {}
+  ...
+}
+```
+
+This becomes very useful for endpoints serving [MediaPipe Graphs](./mediapipe.md). In unary inference RPC each request initializes new MediaPipe graph, inputs are deserialized into MediaPipe packets and pushed into the graph. After the graph is done, it is deleted.
+However, in streaming inference RPC MediaPipe graph is created only once per connection, and is reused by subsequent requests to the same gRPC stream. This removes graph initialization overhead and increases its overall throughput.
+
+![diagram](streaming_diagram.svg)
+
+## Graph Selection
+Opening stream is not enough to create execution graph. Instead, the first gRPC request after the opening defines which graph definition will be selected for execution (`model_name` and `model_version` proto fields).
+Afterwards, subsequent requests are required to match the servable name and version, otherwise the error is reported and input packets are not pushed to the graph. However, the graph remains in available for correct requests.
+
+> NOTE: The server closes the stream after the first request if requested graph is non-existent or retired.
+
+## Timestamping
+MediaPipe Graphs require packets to include timestamp information for synchronization purposes. Each input stream in the graph requires timestamps to be monotonically increasing. Read further about [MediaPipe timestamping](https://developers.google.com/mediapipe/framework/framework_concepts/synchronization#timestamp_synchronization).
+
+### Automatic timestamping
+By default OpenVINO Model Server assigns timestamps automatically. Each gRPC request is treated as separate point on the timeline, starting from 0. Each request is deserialized sequentially and increases the timestamp by 1.
+
+> NOTE: It means that in order to send multiple inputs with the same timestamp, client needs to pack it into single request (or use manual timestamping).
+
+### Manual timestamping
+Optionally, the client is allowed to include timestamp manually via request parameter `OVMS_MP_TIMESTAMP`. It is applied to all the packets deserialized from the request.
+
+It is possible to mix manual/automatic timestamping. After correct deserialization step, default automatic timestamp is always equal to `previous_timestamp + 1`.
+
+## Preserving State Between Requests
+Note that subsequent requests in a stream have access to the same instance of MediaPipe graph. It means that it is possible implement graph that saves intermediate state and act in stateful manner. It might be an advantage f.e. for object tracking use cases.
+
+## Useful links
+- example client snippets TODO
+- [complete demo with streaming](../demos/mediapipe/holistic_tracking/README.md)
```

Comment:
Here we could use direct link to paragraph

Reply:
Will add once @bstrzele adds streaming variant